packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,564 @@
+
+class _CoreClientMock extends Mock implements core.$Client {}
+
+class _AppPasswordClientMock extends Mock implements core.$AppPasswordClient {}
+
+class _ClientFlowLoginV2ClientMock extends Mock implements core.$ClientFlowLoginV2Client {}
+
+class _UsersClientMock extends Mock implements provisioning_api.$UsersClient {}
+
+class _AccountStorageMock extends Mock implements AccountStorage {}
+
+typedef _AccountStream = ({BuiltList<Account> accounts, String? active});
+
+void main() {
+  late AccountStorage storage;
+  late AccountRepository repository;
+
+  late core.$Client coreClient;
+  late core.$AppPasswordClient appPassword;
+  late core.$ClientFlowLoginV2Client clientFlowLoginV2;
+  late provisioning_api.$UsersClient users;
+
+  setUpAll(() {
+    registerFallbackValue(_FakeUri());
+    registerFallbackValue(_FakePollRequest());
+    MockNeonStorage();
+  });
+
+  setUp(() {
+    coreClient = _CoreClientMock();
+    appPassword = _AppPasswordClientMock();
+    clientFlowLoginV2 = _ClientFlowLoginV2ClientMock();
+    users = _UsersClientMock();
+
+    mockedClient = AuthenticationClient(
+      core: coreClient,
+      appPassword: appPassword,
+      clientFlowLoginV2: clientFlowLoginV2,
+      users: users,
+    );
+
+    storage = _AccountStorageMock();
+
+    repository = AccountRepository(
+      userAgent: 'userAgent',
+      httpClient: _FakeClient(),
+      storage: storage,
+    );
+  });
+
+  final accountList = BuiltList<Credentials>([
+    Credentials((b) {
+      b
+        ..serverURL = Uri.https('serverUrl')
+        ..username = 'username'
+        ..password = 'password';
+    }),
+    Credentials((b) {
+      b
+        ..serverURL = Uri.https('other-serverUrl')
+        ..username = 'username'
+        ..password = 'password';
+    }),
+  ]);
+
+  group('AccountRepository', () {
+    test('failure equality', () {
+      expect(
+        FetchStatusFailure(accountList),
+        equals(FetchStatusFailure(accountList)),
+      );
+    });
+
+    group('accounts', () {
+      test('emits empty user map before initialized', () {
+        expect(
+          repository.accounts,
+          emits((accounts: BuiltList<Account>(), active: null)),
+        );
+      });
+    });
+
+    test('hasAccounts', () async {
+      when(() => storage.readAccounts()).thenAnswer((_) async => BuiltList());
+      await repository.loadAccounts();
+
+      expect(repository.hasAccounts, isFalse);
+
+      when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+      await repository.loadAccounts();
+
+      expect(repository.hasAccounts, isTrue);
+    });
+
+    test('accountByID', () async {
+      when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+      await repository.loadAccounts();
+
+      expect(repository.accountByID('invalid'), isNull);
+      expect(repository.accountByID('c73bc81eda746a735de697a0f0cbdb08f96ea3f3'), isNotNull);
```

Comment:
Wouldn't it be better to not hardcode the value and take it from the list? This shouldn't test that the id hashing works, just that you can find accounts by their ids.
Same for all the other places where a hardcoded id is used in the tests.